modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts

```diff
@@ -0,0 +1,264 @@
+type ReliquaryPosition = {
+    amount: BigNumber;
+    rewardDebt: BigNumber;
+    rewardCredit: BigNumber;
+    entry: BigNumber;
+    poolId: BigNumber;
+    level: BigNumber;
+    genesis: BigNumber;
+    lastMaturityBonus: BigNumber;
+};
+
+type BalanceChangedEvent = Event & {
+    args: {
+        pid: BigNumber;
+        amount: BigNumber;
+        to: string;
+        relicId: BigNumber;
+    };
+};
+
+type RelicManagementEvent = Event & {
+    args: {
+        fromId: BigNumber;
+        toId: string;
+        amount: BigNumber;
+    };
+};
+
+type TransferEvent = Event & {
+    args: {
+        from: string;
+        to: string;
+        tokenId: BigNumber;
+    };
+};
+
+export class UserSyncReliquaryFarmBalanceService implements UserStakedBalanceService {
+    constructor() {}
+
+    public async syncChangedStakedBalances(): Promise<void> {
+        const status = await prisma.prismaUserBalanceSyncStatus.findUnique({ where: { type: 'RELIQUARY' } });
+
+        if (!status) {
+            throw new Error('UserReliquaryFarmBalanceService: syncStakedBalances called before initStakedBalances');
+        }
+
+        const pools = await prisma.prismaPool.findMany({ include: { staking: true } });
+        const latestBlock = await jsonRpcProvider.getBlockNumber();
+        const farms = await reliquaryService.getAllFarms({});
+
+        const startBlock = status.blockNumber + 1;
+        const endBlock = latestBlock - startBlock > 10_000 ? startBlock + 10_000 : latestBlock;
+        const amountUpdates = await this.getAmountsForUsersWithBalanceChangesSinceStartBlock(
+            networkConfig.masterchef.address,
+            startBlock,
+            endBlock,
+        );
+        const userAddresses = _.uniq(amountUpdates.map((update) => update.userAddress));
+
+        if (amountUpdates.length === 0) {
+            await prisma.prismaUserBalanceSyncStatus.update({
+                where: { type: 'RELIQUARY' },
+                data: { blockNumber: endBlock },
+            });
+
+            return;
+        }
+
+        await prismaBulkExecuteOperations(
+            [
+                prisma.prismaUser.createMany({
+                    data: userAddresses.map((userAddress) => ({ address: userAddress })),
+                    skipDuplicates: true,
+                }),
+                ...amountUpdates.map((update) => {
+                    const pool = pools.find((pool) => pool.staking?.id === update.farmId);
+                    const farm = farms.find((farm) => farm.id === update.farmId);
+
+                    return prisma.prismaUserStakedBalance.upsert({
+                        where: { id: `reliquary-${update.farmId}-${update.userAddress}` },
+                        update: {
+                            balance: update.amount,
+                            balanceNum: parseFloat(update.amount),
+                        },
+                        create: {
+                            id: `${update.farmId}-${update.userAddress}`,
```

Comment:
need to add `reliquary-` in front of the id string

Reply:
you mean in the create part right?